Initial actions implementation

Pasarus opened the pull request.

martyngigg commented on README.md:
```diff
@@ -1,2 +1,10 @@
-# pre-commit-hooks
-Stores hooks for pre-commit that are not available from the community
+# Pre commit hooks for the MantidProject
```
```suggestion
# Custom pre-commit hooks for the pre-commit framework
```

martyngigg commented on README.md:
```diff
@@ -1,2 +1,10 @@
-# pre-commit-hooks
-Stores hooks for pre-commit that are not available from the community
+# Pre commit hooks for the MantidProject
+The hooks present in this repository are independent and not reliant on Clang format or CppCheck (Not yet implemented)
```
```suggestion
The hooks present in this repository are:
* clang-format (self-contained hook not reliant on clang-format being available externally)
```

Pasarus committed:
Implement clang-format builds and update clang-format


Add actions YAML file


Change auto commit action


Update Auto Commit and add gitignore


Add debug statements to actions for linux


Move to Clang-format dir, update clang format, and add hook config


Update actions to respect the clang-format dir


Change back to the original git-auto-commit-action


Remove bin glob pattern


Re-add bin file pattern, as a sub-directory instead of root


Ensure windows script looks in Release folder


Attempt fastforward pull before committing and only commit .exe


Apply suggestions from code review

Co-authored-by: Martyn Gigg <martyn.gigg@stfc.ac.uk>
Remove pull fast forward


Move mac clang-format to main bin and remove llvm extras


Improve Mac script cleanup

Pasarus committed:
Add executable for Linux

Signed-off-by: mantid-builder <mantid-builder@hosting.com>

Pasarus committed:
Add executable for Windows

Signed-off-by: mantid-builder <mantid-builder@hosting.com>

Pasarus committed:
Add executable for MacOS

Signed-off-by: mantid-builder <mantid-builder@hosting.com>
